server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsNodeResponse.java

```diff
@@ -64,7 +66,12 @@ public ClusterStatsNodeResponse(StreamInput in) throws IOException {
         }
         this.nodeInfo = new NodeInfo(in);
         this.nodeStats = new NodeStats(in);
-        shardsStats = in.readArray(ShardStats::new, ShardStats[]::new);
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
+            this.shardsStats = in.readOptionalArray(ShardStats::new, ShardStats[]::new);
```

Comment:
I have modified the IO to send `optionalArray` instead of a concrete `array` using `readOptionalArray` and `writeOptionalWriteable` instead of `readArray` and `writeArray`.

I did this because we normally always expect this to present. Since now they can be null too, I modified it accordingly. Do you require me to revert back?